docs/investigations/self-service-policy.md

```diff
@@ -0,0 +1,51 @@
+
+
+## Approach:
+
+The logical cluster concept offers an opportunity to allow self-service at a cluster scope, with the effective cost of the namespace partitioning scheme. In addition, the separation of workload at control plane (kcp) and data plane (physical cluster) via transparent multi-cluster or similar schemes allows strong policy control of what configuration is allowed (reject early), restriction of the supported API surface area for workload APIs (limit / control certain fields like pod security), and limits the access of individual users to the underlying infra (much like clusters limit access to nodes).
+
+It should be possible to accomplish current self-service namespace and cluster partitioning via the logical cluster mechanism + policy enforcement, and to incentivize a wider range of "external policy control" users to adopt self-service via stronger control points and desirable use cases (multi-cluster resiliency for apps).
+
+We want to enable concrete points of injection of policy that are difficult today in Kubernetes tenancy:
+
+1. The acquisition of a new logical cluster with **capabilities** and **constraints**
+2. How the APIs in a logical cluster are **transformed** to an underlying cluster
+3. How to manage the evolution of APIs available to a logical cluster over time
+4. New hierarchal policy options are more practical since different logical clusters can have different APIs
+
+
+## Areas of investigation
+
+* Using logical clusters as a mechanism for tenancy, but having a backing implementation that can change
+  * I.e. materialize logical clusters as an API resource in a separate logical cluster
+  * Or implementing logical clusters outside the system and having the kcp server implementation be a shim
+  * Formal "policy module" implementations that can be plugged into a minimal API server while using logical cluster impl
+* Catalog the set of tenancy constructs in use in Kube
+  * Draw heavily on sig-multitenancy explorations - work done by cluster api nested, namespace tenancy, hierarchal namespace designs
```

Comment:
Could you add the link to these design, for example:
[cluster api nested](https://github.com/kubernetes-sigs/cluster-api-provider-nested/tree/main/virtualcluster)

Reply:
Definitely.